libmachine/provision/provisiontest/sshcommander.go

```diff
@@ -0,0 +1,22 @@
+//Package provisiontest provides utilities for testing provisioners
+package provisiontest
+
+import (
+	"errors"
+	"strings"
+)
+
+//FakeSSHCommander is an implementation of provision.SSHCommander to provide predictable responses set by testing code
+//Extend it when needed
+type FakeSSHCommander struct {
+	//Result of the ssh command to look up the FilesystemType
+	FilesystemType string
```

Comment:
> For testing more high level functionality, like checking the default graphdriver in the RedHat provisioner for example, it does not really make sense to know that the underlying call is stat -f -c %T /var/lib, this would also prevent easy refactoring of the specific implementation of the filesystem checking without modifying all tests.

I'm not sure I understand.  If the underlying command sent to SSH changes (say, for Red Hat), I _want_ the tests to break.  We'll then know that the tests need to be updated as well, will have clearly defined expected behavior, and will only need to update one line in the tests.